tools/copyblocks/s3.go

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package main
+
+import (
+	"context"
+	"flag"
+	"io"
+	"strings"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"github.com/pkg/errors"
+)
+
+type s3Config struct {
+	source      s3ClientConfig
+	destination s3ClientConfig
+}
+
+func (c *s3Config) RegisterFlags(f *flag.FlagSet) {
+	c.source.RegisterFlags("s3-source-", f)
+	c.destination.RegisterFlags("s3-destination-", f)
+}
+
+func (c *s3Config) validate(source, destination string) error {
+	if source == serviceS3 {
+		if err := c.source.validate("s3-source-"); err != nil {
+			return err
+		}
+	}
+	if destination == serviceS3 {
+		return c.destination.validate("s3-destination-")
+	}
+	return nil
+}
+
+type s3ClientConfig struct {
+	endpoint  string
+	accessKey string
+	secretKey string
+	secure    bool
+}
+
+func (c *s3ClientConfig) RegisterFlags(prefix string, f *flag.FlagSet) {
+	f.StringVar(&c.endpoint, prefix+"endpoint", "", "The endpoint to contact when accessing the bucket.")
+	f.StringVar(&c.accessKey, prefix+"access-key", "", "The access key used in AWSV4 Authorization.")
+	f.StringVar(&c.secretKey, prefix+"secret-key", "", "The secret key used in AWSV4 Authorization.")
```

Comment:
[nit] AWS refers to it as "AWS Signature v4 authentication" in their docs, would be good to be consistent

Reply:
The classic Authentication vs Authorization blunder, paired with the signature used for authentication passed in the `Authorization` header. I swapped to `AWS Signature Version 4`, thanks!